From 4d1fc446cf152e59adda5e67d693d3fd912e02e2 Mon Sep 17 00:00:00 2001
From: Brennan Lamoreaux <blamoreaux@vmware.com>
Date: Mon, 20 Mar 2023 17:33:50 +0000
Subject: [PATCH] tuned: expand functions for variables in plugin_net

tuned-adm verify fails on realtime profile (and likely others), because
functions are not properly expanded/evaluated in plugin_net.py when read
in from tuned.conf files. Instead, they are simply parsed by delimiters
like a normal parameter.

Fix this issue by properly evaluating config file functions in plugin_net.py.
---
 tuned/plugins/plugin_net.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tuned/plugins/plugin_net.py b/tuned/plugins/plugin_net.py
index 1cc7b17..105646b 100644
--- a/tuned/plugins/plugin_net.py
+++ b/tuned/plugins/plugin_net.py
@@ -329,6 +329,8 @@ class NetTuningPlugin(hotplug.Plugin):
 	# parse features/coalesce config parameters (those defined in profile configuration)
 	# context is for error message
 	def _parse_config_parameters(self, value, context):
+		# expand config variables
+		value = self._variables.expand(value)
 		# split supporting various dellimeters
 		v = str(re.sub(r"(:\s*)|(\s+)|(\s*;\s*)|(\s*,\s*)", " ", value)).split()
 		lv = len(v)
-- 
2.35.6

